src/proxy.rs

```diff
@@ -105,15 +114,43 @@ impl Proxy {
             outbound: self.outbound.address(),
             inbound: self.inbound.address(),
             socks5: self.socks5.address(),
+            dns_proxy: match &self.dns_proxy {
+                None => SocketAddr::new(IpAddr::V4(Ipv4Addr::UNSPECIFIED), 0),
+                Some(dns_proxy) => dns_proxy.address(),
+            },
         }
     }
 }
 
+async fn new_dns_proxy(pi: ProxyInputs) -> Result<Option<dns::DnsProxy>, Error> {
+    // Determine whether we should enable DNS from proxy metadata.
+    const DNS_CAPTURE_METADATA: &str = "ISTIO_META_DNS_CAPTURE";
+    let dns_enabled = pi
+        .cfg
+        .proxy_metadata
```

Comment:
cfg should probably have a `dns_proxy_enabled` (or perhaps `Option<SocketAddr>`) and we just use it directly?

Also I think the istio_meta_ is stripped (https://github.com/istio/ztunnel/blob/17620574ce78abf62b98c8dda2900e55bbae1d1d/src/config.rs#L403)

Reply:
I was hoping to leverage the existing configuration for DNS capture for now, rather than introducing something new. I don't have a strong opinion here.

@bleggett thoughts?

Regarding using proxy metadata (since it seems I'm probably not using it correctly here), do we have other examples in ztunnel that use it? If we have constants defined somewhere, I can add the DNS one there.

Reply:
I don't mean anything user facing, just we currently process all settings into `cfg` struct and now we are parsing them on-demand

Reply:
Ok sure .. just trying to figure out the desired end state here. Are you suggesting that I read the metadata and then set the flag on the config based on that?